app/models/driver.rb

```diff
@@ -0,0 +1,18 @@
+class Driver < ApplicationRecord
+  has_many :trips
+  
+  validates :name, presence: true
+  validates :vin, presence: true
+  validates :available, inclusion: { in: [true, false] }
+
+  def get_total_earnings
+    trips = Trip.where(driver_id: self.id)
+    #The driver gets 80% of the trip cost after a fee of $1.65 is subtracted
+    return trips.empty? ? 0 : trips.map  { |trip| (trip.cost - 1.65) * 0.8 }.sum
```

Comment:
Note that because a driver has many trips, it automatically gets a trips method which returns a collection of trips belonging to the driver. 

Note below the `self.` is optional, just for clarity.

```suggestion
    #The driver gets 80% of the trip cost after a fee of $1.65 is subtracted
    return self.trips.empty? ? 0 : self.trips.map  { |trip| (trip.cost - 1.65) * 0.8 }.sum
```